### PR TITLE
Migrate state-db, state-machine, telemetry, test-client and test-runtime to the 2018 edition

### PR DESCRIPTION
**Pull request by DarkEld3r**
Part of the #1415 issue.

### CODE REVIEW

File: core/state-db/src/noncanonical.rs [hunk 1]
```diff
@@ -270,9 +272,9 @@ impl<BlockHash: Hash, Key: Hash> NonCanonicalOverlay<BlockHash, Key> {
 mod tests {
 	use std::io;
 	use super::NonCanonicalOverlay;
-	use {ChangeSet};
+	use crate::{ChangeSet};
```
Comment (bkchr) [inline core/state-db/src/noncanonical.rs]:
```suggestion
	use crate::ChangeSet;
```

File: core/state-db/src/lib.rs [hunk 2]
```diff
@@ -29,23 +29,17 @@
 //! See `RefWindow` for pruning algorithm details. `StateDb` prunes on each canonicalization until pruning
 //! constraints are satisfied.
 
-#[macro_use] extern crate log;
-#[macro_use] extern crate parity_codec_derive;
-extern crate parking_lot;
-extern crate parity_codec as codec;
```
Comment (bkchr) [inline core/state-db/src/lib.rs]: As we can not rename `parity_codec` currently in the `Cargo.toml`, I would like to stick to renaming it here.

File: core/state-db/src/pruning.rs [hunk 3]
```diff
@@ -158,8 +160,8 @@ impl<BlockHash: Hash, Key: Hash> RefWindow<BlockHash, Key> {
 mod tests {
 	use super::RefWindow;
 	use primitives::H256;
-	use {CommitSet};
-	use test::{make_db, make_commit, TestDb};
+	use crate::{CommitSet};
```
Comment (bkchr) [inline core/state-db/src/pruning.rs]:
```suggestion
	use crate::CommitSet;
```

File: core/state-db/src/noncanonical.rs (hunk 1 above)
Comment (DarkEld3r) [reply core/state-db/src/noncanonical.rs]: I also prefer to omit braces in such cases, but I don't want to add unnecessary changes.
Fixed.

**Pull request merged**